apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -183,9 +183,13 @@ impl Drop for Telemetry {
         ::tracing::debug!("dropping telemetry...");
         let count = TELEMETRY_REFCOUNT.fetch_sub(1, Ordering::Relaxed);
         if count < 2 {
-            std::thread::spawn(|| {
+            // We don't want telemetry to drop until the shutdown completes.
+            // Ignore the result, we log errors as warnings.
+            let _ = std::thread::spawn(|| {
                 opentelemetry::global::shutdown_tracer_provider();
-            });
+            })
+            .join()
+            .map_err(|e| ::tracing::warn!("unable to join tracer shutdown thread: {:?}", e));
```

Comment:
I'm wondering if we should put a timeout on this ?

Reply:
That's a good suggestion. I could add a variation on this: https://stackoverflow.com/questions/36181719/what-is-the-correct-way-in-rust-to-create-a-timeout-for-a-thread-or-a-function
I'll take a look at doing that.
Only thing is knowing how long to wait... (I'll pick 5 seconds for now)

Reply:
starting a thread to immediately join on it is the same as calling the code directly. The thread is not necessary (unless you add the timeout)

Reply:
I agree. I had a conversation with Bryn about why we even needed the thread at all... Anyway, I'm updating it to add a timeout as per @bnjjj suggestion.

Reply:
done: https://github.com/apollographql/router/pull/2191/commits/7ca2897d3264df7d049d8f8cc1cccccac569924d